frame/support/src/storage/migration.rs

```diff
@@ -389,6 +389,39 @@ pub fn move_prefix(from_prefix: &[u8], to_prefix: &[u8]) {
 	}
 }
 
+/// A phony migration that does nothing, except executing `TryDecodeEntireStorage` on
+/// `post_upgrade`, which implies it is only available if `try-state` feature is used.
+///
+/// This can be used typically in the top level runtime, where `AllPallets` typically comes from
+/// `construct_runtime!`.
+pub struct EnsureStateDecodes<AllPallets>(sp_std::marker::PhantomData<AllPallets>);
+
+#[cfg(not(feature = "try-runtime"))]
+impl<AllPallets> crate::traits::OnRuntimeUpgrade for EnsureStateDecodes<AllPallets> {
+	fn on_runtime_upgrade() -> crate::weights::Weight {
+		Default::default()
+	}
+}
+
+#[cfg(feature = "try-runtime")]
+impl<AllPallets: crate::traits::TryDecodeEntireStorage> crate::traits::OnRuntimeUpgrade
+	for EnsureStateDecodes<AllPallets>
+{
+	fn on_runtime_upgrade() -> sp_weights::Weight {
+		Default::default()
+	}
+
+	fn post_upgrade(_: Vec<u8>) -> Result<(), &'static str> {
+		let decoded = AllPallets::try_decode_entire_state()?;
```

Comment:
interestingly, this breaks every time we want to do a lazy migration, since a lazy migration is by definition an undecode-able state. example of this is https://github.com/paritytech/substrate/commit/f8a1dd4c836d7d29382ba66fc2fce40fc1364546

In the future, we should add a `expected: Vec<Vec<u8>>` to the call, which is a list of expected prefixes that is allowed to have un-decode-able values in it.